drivers/gles2/shader_compiler_gles2.cpp

```diff
@@ -204,6 +204,7 @@ static String get_constant_text(SL::DataType p_type, const Vector<SL::ConstantNo
 		} break;
 		default: ERR_FAIL_V(String());
 	}
+	ERR_FAIL_V(String());
```

Comment:
Isn't this the kind of return that we actually *removed* in previous PRs to satisfy static checks? I don't see any branch which might not return and thus this is unreachable code IINM.

Reply:
Yes, it was removed and during compilation without any parameters or with almost all available, everything worked.
I think the GCC 8.3 has problems with the tsan option and badly informs about the missing return, and in Clang 8.0 everything works well.

Reply:
CC @marxin - is this worth investigating further?

Reply:
Yes!

I can't see it, both GCC 8 and are fine for me:
```
marxin@marxinbox:~/Programming/godot> g++ -o modules/gdscript/gdscript_parser.x11.opt.tools.64s.o -c -std=c++11 -g1 -O2 -fsanitize=thread -pipe -fpie -Wall -Wshadow-local -Werror -DDEBUG_ENABLED -DTOUCH_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -D_REENTRANT -DJOYDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DOPENGL_ENABLED -DGLES_ENABLED -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -DGLAD_ENABLED -DGLES_OVER_GL -DMODULE_ASSIMP_ENABLED -DMODULE_BMP_ENABLED -DMODULE_BULLET_ENABLED -DMODULE_CSG_ENABLED -DMODULE_CVTT_ENABLED -DMODULE_DDS_ENABLED -DMODULE_ENET_ENABLED -DMODULE_ETC_ENABLED -DMODULE_FREETYPE_ENABLED -DMODULE_GDNATIVE_ENABLED -DMODULE_GDSCRIPT_ENABLED -Ithirdparty/libpng -Ithirdparty/glad -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/x11 -I. -Ieditor modules/gdscript/gdscript_parser.cpp
marxin@marxinbox:~/Programming/godot> g++-8 -o modules/gdscript/gdscript_parser.x11.opt.tools.64s.o -c -std=c++11 -g1 -O2 -fsanitize=thread -pipe -fpie -Wall -Wshadow-local -Werror -DDEBUG_ENABLED -DTOUCH_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -D_REENTRANT -DJOYDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DOPENGL_ENABLED -DGLES_ENABLED -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -DGLAD_ENABLED -DGLES_OVER_GL -DMODULE_ASSIMP_ENABLED -DMODULE_BMP_ENABLED -DMODULE_BULLET_ENABLED -DMODULE_CSG_ENABLED -DMODULE_CVTT_ENABLED -DMODULE_DDS_ENABLED -DMODULE_ENET_ENABLED -DMODULE_ETC_ENABLED -DMODULE_FREETYPE_ENABLED -DMODULE_GDNATIVE_ENABLED -DMODULE_GDSCRIPT_ENABLED -Ithirdparty/libpng -Ithirdparty/glad -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/x11 -I. -Ieditor modules/gdscript/gdscript_parser.cpp
```

Reply:
@qarmin Can you provide a pre-processed source file (`-E` option) and the full command line?

Reply:
My command
```
g++ -o modules/gdscript/gdscript_parser.x11.tools.64s.o -c -std=c++11 -g3 -fsanitize=thread -pipe -fpie -Wall -Wshadow-local -Werror=return-type -DDEBUG_ENABLED -DDEBUG_MEMORY_ENABLED -DDEBUG_MEMORY_ALLOC -DDISABLE_FORCED_INLINE -DTOUCH_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -D_REENTRANT -DJOYDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DOPENGL_ENABLED -DGLES_ENABLED -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -DGLAD_ENABLED -DGLES_OVER_GL -DMODULE_ASSIMP_ENABLED -DMODULE_BMP_ENABLED -DMODULE_BULLET_ENABLED -DMODULE_CSG_ENABLED -DMODULE_CVTT_ENABLED -DMODULE_DDS_ENABLED -DMODULE_ENET_ENABLED -DMODULE_ETC_ENABLED -DMODULE_FREETYPE_ENABLED -DMODULE_GDNATIVE_ENABLED -DMODULE_GDSCRIPT_ENABLED -Ithirdparty/libpng -Ithirdparty/glad -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/x11 -I. -Ieditor modules/gdscript/gdscript_parser.cpp -E
```
[file.txt](https://github.com/godotengine/godot/files/3476986/file.txt)

Scons command
```
scons p=x11 -j6 use_tsan=yes verbose=yes
```

Reply:
Thanks for it, there's un upstream issue for it:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=91389

Reply:
And the conclusion is that Godot should clean up code so that it does not contain so much dead code in the switch statements: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=91389#c5

Reply:
OK, the warning location is pretty weird though if the problem is having `break;`s after `returns;` further up in the switch.

And the package maintainers "weird convention" is just an attempt to keep C++'s weird implicit fallthrough feature manageable, as ensuring that a case ends either with `break;` or an explicit fallthrough helps prevent issues.

But in this case I guess we can remove the `break;`s indeed.

Reply:
> But in this case I guess we can remove the `break;`s indeed.

Or you can move the `return` from `default` case and do it at the very end of the function.